Preserve the SIMPLE_CI environment variable when executing the tests as user 'ohpc'

martin-g opened the pull request:
Add mapping for FFTW test.
Fix the value of TASKS depending on whether or not SIMPLE_CI is exported.

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Preserve the SIMPLE_CI environment variable when executing the tests as user ohpc

Add mapping for FFTW test.
Fix the value of TASKS depending on whether or not SIMPLE_CI is
exported.

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
RHEL and Rocky (8.6) do not support `-w`:
```
su --help

Usage:
 su [options] [-] [<user> [<argument>...]]

Change the effective user ID and group ID to that of <user>.
A mere - implies -l.  If <user> is not given, root is assumed.

Options:
 -m, -p, --preserve-environment  do not reset environment variables
 -g, --group <group>             specify the primary group
 -G, --supp-group <group>        specify a supplemental group

 -, -l, --login                  make the shell a login shell
 -c, --command <command>         pass a single command to the shell with -c
 --session-command <command>     pass a single command to the shell with -c
                                   and do not create a new session
 -f, --fast                      pass -f to the shell (for csh or tcsh)
 -s, --shell <shell>             run <shell> if /etc/shells allows it
 -P, --pty                       create a new pseudo-terminal

 -h, --help                      display this help
 -V, --version                   display version
```

I will rework it to use `sudo` instead.

martin-g committed:
Use 'sudo` instead of 'su' to be able to preserve env vars

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Add a helper function that returns the number of parallel tasks to run

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Use tasks_count() helper function for most tests

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented on tests/ci/spec_to_test_mapping.py:
```diff
@@ -42,6 +42,10 @@
+    '../components/parallel-libs/fftw/SPECS/fftw.spec': [
+        'fftw',
+        'openmpi4-gnu12-ohpc mpich-gnu12-ohpc'
+    ],
```
Should I add entries also for all tests whose `rm_execution` has been modified below ?

martin-g replied:
I'll do it!

martin-g commented on tests/ci/spec_to_test_mapping.py:
```diff
@@ -42,6 +42,10 @@
         'scotch',
         'zlib-devel'
     ],
+    '../components/parallel-libs/fftw/SPECS/fftw.spec': [
```
The leading `../` will be removed in the next commit!
It seems this typo didn't break the test execution ...

martin-g replied:
Answering to myself: If the key is misspelled then `setup_slurm_and_run_tests.sh` cannot find it and it does not execute any tests.

martin-g commented on tests/mpi/tests/rm_execution_multi_host:
```diff
@@ -12,7 +12,7 @@ check_rms
 
 NODES=2
-TASKS=8
+TASKS=`tasks_count 8`
```
Which .spec should be added to `spec_to_test_mapping.py` for this test ?

adrianreber replied:
see the other mpi test

martin-g commented on tests/mpi/tests/rm_execution_single_host:
```diff
@@ -12,7 +12,7 @@ check_rms
 
 NODES=1
-TASKS=2
+TASKS=`tasks_count 2`
```
Which .spec should be added to `spec_to_test_mapping.py` for this test ?

adrianreber replied:
probably openmpi4.spec, mpich.spec and mvapich2.spec

martin-g replied:
I've added mappings for `openmpi4` and `mpich`.
`mvapich2` requires InfiniBand and thus cannot be tested on Github Actions.

martin-g commented on tests/rms-harness/tests/test_harness:
```diff
@@ -12,7 +12,7 @@ check_rms
 
 NODES=2
-TASKS=2
+TASKS=`tasks_count 2`
```
Which .spec should be added to spec_to_test_mapping.py for this test ?

adrianreber replied:
maybe always?

martin-g replied:
I.e. to add `--enable-rms-harness` at https://github.com/martin-g/openeuler-ohpc/blob/preserve_SIMPLE_CI_for_test_execution/tests/ci/setup_slurm_and_run_tests.sh#L92 ?

adrianreber replied:
Yes, something like this. Not sure myself.

martin-g replied:
Done!
We can remove it if it causes any problems!

martin-g replied:
It failed on RHEL.
It passes fine here (on openEuler).
I'm trying to find the reason.

martin-g replied:
The error is:
```
-------------------------------------------------------
User Environment: RMS Harness tests: gnu12-openmpi4
-------------------------------------------------------
Lmod has detected the following error: The following module(s) are unknown:
"prun"
```

martin-g committed:
Add more test mappings for all spec whose rm_execution has been modified in this PR

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Add mappings for openmpi4 and mpich

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Always enable rms-harness

Suggested by @adrianreber at https://github.com/openhpc/ohpc/pull/1638#discussion_r1052181670

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Print the log files in case of failure in the tests

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Always install prun-ohpc

It is needed by rms-harness tests

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Pre-install gnu12-compilers-ohpc for the tests

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Pre-install openmpi4-gnu12-ohpc and mpich-gnu12-ohpc

They are needed by rms-harness, and by all other modules.

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented on tests/ci/setup_slurm_and_run_tests.sh:
```diff
@@ -12,14 +12,15 @@ dnf -y install \
 # Install rebuilt packages (if any)
 # shellcheck disable=SC2046 # (we want the words to be split)
-dnf -y install $(find /home/"${USER}"/rpmbuild/RPMS/ -name "*rpm") || true
+dnf -y install prun-ohpc gnu12-compilers-ohpc openmpi4-gnu12-ohpc mpich-gnu12-ohpc $(find /home/"${USER}"/rpmbuild/RPMS/ -name "*rpm") || true
```
Those are needed by RMS Harness.
Is it OK to pre-install them here ?

adrianreber commented:
That is a big PR and I hope it does not break anything. It looks good from the CI point of view. I will merge it. Thanks.
